_utils/spellcheck.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+v() { [ -n "$VERBOSE" ] && echo "$@" ; }
+
+if command -v hunspell > /dev/null; then
+    check_words() { hunspell -l; }
+elif command -v aspell > /dev/null; then
+    check_words() { aspell | grep '^[#&]' | cut -d' ' -f2; }
+else
+    echo "${0##*/}: No spell checker installed. Need hunspell or aspell." >&2
+    exit 1
+fi
+
+extract_words() {
+    pandoc -tjson "$@" | jq -r '
+    # will be builtin in a future version of jq. is already in master
+    # Apply f to composite entities recursively, and to atoms
+    def walk(f):
+      . as $in
+      | if type == "object" then
+          reduce keys[] as $key
+            ( {}; . + { ($key):  ($in[$key] | walk(f)) } ) | f
+      elif type == "array" then map( walk(f) ) | f
+      else f
+      end;
+
+    # strip code
+    walk (
+      if (type == "object") then
+        if (.t == "CodeBlock" or .t == "Code") then
+          empty
+        else
+          .
+        end
+      else
+        .
+      end
+    ) |
+
+    # extract raw string nodes
+    .. | select(type=="object" and has("t") and .t=="Str") | .c
+    '
+}
+
+find_suspects() {
+    local workdir=$1
+    local outdir=$2
+    mkdir -p "$outdir"
+    v "Parsing $outdir..."
+    find "$workdir" -mindepth 1 -type d -printf '%P\0' \
+      | ( cd "$outdir" && xargs -0 mkdir -p ) >&2
+    find "$workdir" -type f -name '*.html' -printf '%P\0' \
+      | while read -d '' f; do
+        extract_words "${workdir}/${f}" | check_words | sort -u \
+          > "${outdir}/${f}.wrong"
+    done
+}
+
+check_one() {
+    local ref=$1
+    local checkout="${d}/${ref}.checkout"
+    [ -d "$checkout" ] && { v "Using cached $ref"; return; }
+    local tree="${d}/${ref}.suspects.tree"
+    local flat="${d}/${ref}.suspects.flat"
+    v "Checking out $ref..."
+    mkdir -p "$checkout"
+    git --work-tree="$checkout" checkout "$ref" -- .
+    v "Building $ref..."
+    ( cd "$checkout" && make > /dev/null )
+    find_suspects "$checkout/_site" "$tree"
+    find "$tree" -type f -exec cat {} + | sort -u > "$flat"
+}
+
+diff_two() {
+    local old=$1
+    local new=$2
+    local cmp="${d}/${old}..${new}"
+    check_one "$old"
+    check_one "$new"
+    [ -e "${cmp}.plusminus" ] && { v "Using cached diff"; return; }
+    diff -urd "$d"/{"${old}","${new}"}.suspects.tree > "${cmp}.diff.tree" || :
```

Comment:
Yes, unless you want to read it manually (with `SPELL_CHECK_CACHE` set). The fact I didn't notice I should have also had -N means this apparently isn't actually all that useful. Removing.